### DISCUSSION TITLE
Struggling with ESPHome and Mopeka integration

### DISCUSSION THREAD

**Discussion by mac-za** [spbrogan/sensor.mopeka_pro_check#30] [Q&A]
Good Afternoon Sean

I found your integration after doing research on the Mopeka tank check module

Everything worked without any issues.  After changing the LPG tank, the "BLE connection" never seems to be made. Using my phone and the tank check app, I always receive the data. Initially it seemed to be location specific to get it to work, but it seems that the window of scanning misses the window of transmission.  The Mopeka module broadcasts every 3.5 seconds at the moment

I have played around with the Interval and Duration settings, but still no luck. 

What am I doing wrong?

I currently use a D1 Mini ESP32

The code snippet is below:

esphome:
  name: lpg-monitor

esp32:
  #board: esp32dev
  board: wemos_d1_mini32
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: xxxxxxx

ota:
  password: xxxxx

wifi:
  ssid: xxxx
  password: xxxx
  fast_connect: true

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Lpg-Monitor Fallback Hotspot"
    password: xxxx

captive_portal:

esp32_ble_tracker:
  scan_parameters:
    active: True
    interval: 320ms
    window: 60ms
        
sensor:
  # Example using 48KG vertical LPG tank.
  - platform: mopeka_pro_check
    mac_address: xxxx
    tank_type: CUSTOM
    custom_distance_full: 80cm
    custom_distance_empty: 10mm
    temperature:
        name: "LPG Cylinder Temperature"
        #accuracy_decimals: 1
    level:
        name: "LPG level"
        #accuracy_decimals: 1
    distance:
        name: "LPG distance"
        #accuracy_decimals: 1
    battery_level:
        name: "LPG battery level"
        #accuracy_decimals: 1

Thanks in advance

Kenneth

**Comment by spbrogan** [comment]
Kenneth,
Interesting.  I haven't seen this problem or heard of others with the issue.  I know that isn't helpful.  :)  
The range can be somewhat limited and probably has worse range than your cellphone.  Do you think that could be it?  Otherwise, an esphome log would be helpful to see if there is anything interesting/odd in there.  

Let me know if you can collect and share one of those.  
Thanks
Sean
